.changeset/slow-items-heal.md

```diff
@@ -0,0 +1,38 @@
+---
+"astro": minor
+---
+
+Adds experimental JSON Schema support for content collections of with `type: 'data'`.
+
+To enable this feature, add the experimental flag:
```

Comment:
```suggestion
Adds experimental JSON Schema support for content collections of with `type: 'data'`.

This feature will auto-generate a content collection JSON `$schema` for TypeScript-style autocompletion/hints in tools like VSCode.

To enable this feature, add the experimental flag:
```
Make this sentence accurate, but I think some kind of line describing what this is or why it's useful would work here! (I just pulled this from the [roadmap discussion description](https://github.com/withastro/roadmap/discussions/696), but of course this feature might not do exactly that! Write whatever makes sense here, and I'll come back and look at it!)